includes/wikia/Wikia.php

```diff
@@ -1599,6 +1601,20 @@ static public function onSkinTemplateOutputPageBeforeExec(SkinTemplate $skinTemp
 		$tpl->set( 'thisurl', $title->getPrefixedURL() );
 		$tpl->set( 'thisquery', $skinTemplate->thisquery );
 
+		if( !empty( $wgDevelEnvironment ) || !empty( $wgStagingEnvironment ) ) {
+			$out->setRobotPolicy( $wgDefaultRobotPolicy );
+		}
+
+		if( !empty($stagingHeader) ) {
+		// we've got special cases like externaltest.* and showcase.* aliases:
+		// https://github.com/Wikia/wikia-vcl/blob/master/wikia.com/control-stage.vcl#L15
+		// those cases for backend look like production,
+		// therefore we don't want to base only on environment variables
+		// but on HTML headers as well, see:
+		// https://github.com/Wikia/app/blob/dev/redirect-robots.php#L285
+			$out->setRobotPolicy( 'noindex,nofollow' );
```

Comment:
Yeah, this is tricky. It's connected to [the config changes](https://github.com/Wikia/config/pull/1261#discussion-diff-38101957R2939) which sets `$wgDefaultRobotPolicy` to `'noindex,nofollow'`. So, this line is only for those "special cases" like domain aliases `showcase.*` and `externalcase.*` (which is not really a case since it's protected with basic apache authentication).